render/drm_format_set.c

```diff
@@ -137,3 +137,37 @@ struct wlr_drm_format *wlr_drm_format_dup(const struct wlr_drm_format *format) {
 	memcpy(duped_format, format, format_size);
 	return duped_format;
 }
+
+struct wlr_drm_format *wlr_drm_format_intersect(
+		const struct wlr_drm_format *a, const struct wlr_drm_format *b) {
+	assert(a->format == b->format);
+
+	size_t format_len = a->len < b->len ? a->len : b->len;
+	size_t format_size = sizeof(struct wlr_drm_format) +
+		format_len * sizeof(a->modifiers[0]);
+	struct wlr_drm_format *format = calloc(1, format_size);
+	if (format == NULL) {
+		wlr_log_errno(WLR_ERROR, "Allocation failed");
+		return NULL;
+	}
+	format->format = a->format;
```

Comment:
What is `format->cap` and does it have to be zero in all the cases? (I'd have thought that if `*a == *b`, then we expect `*a == *format` in the end)

Reply:
That's a good point. `format->cap` is set appropriately now.